lib/services/flowdock.ts

```diff
@@ -16,295 +16,83 @@ limitations under the License.
 
 import * as Promise from 'bluebird';
 import { Session } from 'flowdock';
-import * as _ from 'lodash';
 import * as path from 'path';
-import * as request from 'request-promise';
-import { FlowdockConstructor, FlowdockEmitContext, FlowdockHandle, FlowdockMessage } from './flowdock-types';
-import { Messenger } from './messenger';
 import {
-	DataHub, MessengerAction, MessengerEmitResponse, MessengerEvent, ReceiptContext, TransmitContext
-} from './messenger-types';
+	FlowdockConnectionDetails, FlowdockEmitContext,
+	FlowdockHandle, FlowdockResponse
+} from './flowdock-types';
 import { ServiceEmitter, ServiceListener } from './service-types';
+import { ServiceUtilities } from './service-utilities';
 
-export class FlowdockService extends Messenger implements ServiceEmitter, ServiceListener, DataHub {
+export class FlowdockService extends ServiceUtilities<string> implements ServiceEmitter, ServiceListener {
 	private static _serviceName = path.basename(__filename.split('.')[0]);
 	private session: Session;
-	private data: FlowdockConstructor;
+	private org: string;
 
-	public constructor(data: FlowdockConstructor, listen = true ) {
-		super(listen);
-		this.data = data;
+	constructor(data: FlowdockConnectionDetails, listen: boolean) {
```

Comment:
As per Front comments, `listen` could be an option in `FlowdockConnectionDetails`.